contracts/modules/cover/Cover.sol

```diff
@@ -437,15 +439,37 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon {
   // TODO: implement properly. we need the staking interface for burning.
   function performPayoutBurn(
     uint coverId,
-    uint /*segmentId*/,
-    uint amount
+    uint segmentId,
+    uint burnAmount
   ) external onlyInternal override returns (address /* owner */) {
 
     ICoverNFT coverNFTContract = ICoverNFT(coverNFT);
     address owner = coverNFTContract.ownerOf(coverId);
 
     CoverData storage cover = _coverData[coverId];
-    cover.amountPaidOut += SafeUintCast.toUint96(amount);
+    cover.amountPaidOut += SafeUintCast.toUint96(burnAmount);
+
+    CoverSegment memory segment = coverSegments(coverId, segmentId);
+    PoolAllocation[] storage allocations = coverSegmentAllocations[coverId][segmentId];
+
+    // burn an
+    uint burnRatio = burnAmount * BURN_DENOMINATOR / segment.amount;
```

Comment:
denominator - mathematics : the part of a fraction that is below the line and that functions as the divisor of the numerator

BURN_DENOMINATOR is not a denominator but a precision factor. I've simply named the constant PRECISION whenever I had divisions between NXM amounts.

Reply:
solved by removing that variable entirely